src/software/backend/simulation/physics/physics_simulator.h

```diff
@@ -49,14 +48,31 @@ class PhysicsSimulator
     PhysicsSimulator(const PhysicsSimulator&)            = delete;
 
     /**
-     * Advances the physics simulation by the given time step, and returns the new state
-     * of the world
+     * Advances the physics simulation by the given time step
      *
      * @param time_step How much to advance the simulation by
+     */
+    void stepSimulation(const Duration& time_step);
+
+    /**
+     * Returns the current state of the world
+     *
+     * @return the current state of the world
+     */
+    World getWorld() const;
+
+    // TODO: Implement std::weak_ptr<SimulatorWorld> getSimulatorWorld and replace this
+    /**
+     * Returns the PhysicsRobots being used in this simulator.
+     *
+     * We return weak pointers because this class owns the physics world the robots are a
```

Comment:
ifeelbadevenpointingthisoutnit: can we join physics and world to PhysicsWorld it was hard to read separately

Reply:
I'm working on this right now

Reply:
Yea @jared-paul is doing PhysicsWorld, will be merged separately. Once it is I'll join stuff up.